draft-irtf-cfrg-opaque.md

```diff
@@ -802,21 +731,25 @@ Steps:
 2. M = Deserialize(request.data)
 3. Z = Evaluate(kU, M)
 4. data = Z.encode()
-5. Create RegistrationResponse response with (request.context, data)
+5. Create RegistrationResponse response with (data)
 6. Output (response, kU)
 ~~~
 
 #### FinalizeRequest
 
 ~~~
-FinalizeRequest(IdU, PwdU, skU, metadata, request, response)
+FinalizeRequest(IdU, PwdU, aad, skU, metadata, request, response)
 
 Parameters:
 - params, the MHF parameters established out of band
+- Nn, length of the key derivation nonce
+- Nk, length of the authentication and exporter keys
 
 Input:
 - IdU, an opaque byte string containing the user's identity
 - PwdU, an opaque byte string containing the user's password
+- aad, application-specific additional associated data
```

Comment:
what is the use of aad (additional to auth_data)?

Reply:
Any AAD specific to the application, really. It's an opaque slot.